packages/dx-scheduler-core/src/types/resources.types.ts

```diff
@@ -0,0 +1,90 @@
+import { Color as MUIColor } from '@material-ui/core';
+import { PureComputed } from '@devexpress/dx-core';
+import { Appointment } from './../index';
+
+/* tslint:disable no-empty-interface */
+/** Specifies a resource that available in scheduler. */
+export type Resource = {
+  /** The name of the appointment object field that specifies a resource of this kind. */
+  fieldName: string;
+  /** Specifies the resource title. */
+  title?: string;
+  /** Indicates whether or not several resources of this kind can be assigned to an appointment. */
+  allowMultiple?: boolean;
+  /** Specifies an array of resource item objects. */
+  items: Array<ResourceItem>;
+};
+
+/** Specifies the resource item object. */
+export type ResourceItem = {
+  /**
+   * Specifies a resource item's identifier.
+   * This id should be related with scheduler's data `fieldName` id.
+   */
+  id: number | string;
+  /**
+   * Specifies resource item's color.
+   * It is used to indicate appointments related to this resource.
+   */
+  color?: string | Color;
+  /** Specifies the resource item's text that would be used in UI */
+  text?: string;
+};
+/** Specifies the resources palette. */
+export type Palette = Array<string | Color>;
+
+/** Specifies the color object. */
+export interface Color extends MUIColor {}
+
+/** The resource item with all properties. */
+export type ValidResourceItem = {
+  /**
+   * Specifies a resource item's identifier.
+   * This id should be related with scheduler's data `fieldName` id.
+   */
+  id: number | string;
+  /**
+   * Specifies resource item's color.
+   * It is used to indicate appointments related to this resource.
+   */
```

Comment:
I think it's not good to reference theme in a core package. Do we need mui color?

Reply:
I think yeah, we need it. Integration with Material-UI is our main (only one) advantage.
This scheduler has only one theme. I think we can close our eyes and do it.

Reply:
This comment addresses this line:
```export interface Color extends MUIColor {}```

Reply:
Ok, let it be so :)